batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/participants/BatchArtifactRenameParticipant.java

```diff
@@ -0,0 +1,204 @@
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.CompositeChange;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.TextFileChange;
+import org.eclipse.ltk.core.refactoring.participants.CheckConditionsContext;
+import org.eclipse.ltk.core.refactoring.participants.RenameParticipant;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.text.edits.MultiTextEdit;
+import org.eclipse.text.edits.ReplaceEdit;
+import org.eclipse.text.edits.TextEdit;
+import org.jboss.tools.batch.core.BatchConstants;
+import org.jboss.tools.batch.core.IBatchArtifact;
+import org.jboss.tools.batch.core.IBatchProject;
+import org.jboss.tools.batch.internal.core.impl.BatchProject;
+import org.jboss.tools.batch.internal.core.impl.BatchProjectFactory;
+import org.jboss.tools.batch.internal.core.impl.BatchUtil;
+import org.jboss.tools.batch.ui.JSTJobUiPlugin;
+import org.jboss.tools.common.java.IAnnotationDeclaration;
+import org.jboss.tools.common.refactoring.FileChangeFactory;
+import org.jboss.tools.common.text.ITextSourceReference;
+import org.jboss.tools.common.util.BeanUtil;
+
+public class BatchArtifactRenameParticipant extends RenameParticipant {
+	private IType type;
+	private String newName;
+	private RefactoringStatus status;
+	private CompositeChange rootChange;
+	private TextFileChange lastChange;
+	private ArrayList<String> keys = new ArrayList<String>();
+
+	@Override
+	protected boolean initialize(Object element) {
+		if (element instanceof IType) {
+			type = (IType) element;
+
+			status = new RefactoringStatus();
+			rootChange = new CompositeChange(BatchParticipantMessages.Updating_Batch_Artifacts_References);
+
+			return true;
+		}
+		return false;
+	}
+
+	@Override
+	public String getName() {
+		return null;
+	}
+
+	@Override
+	public RefactoringStatus checkConditions(IProgressMonitor pm, CheckConditionsContext context)
+			throws OperationCanceledException {
+
+		IProject project = type.getJavaProject().getProject();
+
+		BatchProject batchProject = (BatchProject) BatchProjectFactory.getBatchProjectWithProgress(project);
+
+		if (batchProject == null) {
+			return status;
+		}
+
+		IBatchArtifact artifact = batchProject.getArtifact(type);
+		if (artifact != null) {
+			IAnnotationDeclaration declaration = artifact.getNamedDeclaration();
+			if (declaration == null || declaration.getMemberValue(null) != null) {
+				return status;
+			}
+			newName = BeanUtil.getDefaultBeanName(getArguments().getNewName());
+		} else {
+			newName = type.getFullyQualifiedName().replace(type.getElementName(), getArguments().getNewName());
```

Comment:
What if you are renaming "org.domain.test.test" -> "org.domain.test.BatchTest". Then the newName == "org.domain.BatchTest.BatchTest" which is not correct. It should be "org.domain.test.BatchTest".